src/IterTools.jl

```diff
@@ -1031,4 +1028,58 @@ function iterate(fs::FieldValues, state=1)
     return (getfield(fs.x, state), state + 1)
 end
 
+# IMerge
+
+"""
+    imerge(a,b, predicate = <=, fa = identity, fb = identity)
+
+Iterate over the union of `a` and `b`, merge-sort style.
+
+Input:
+ - `predicate(ak,bk) -> Bool`:
+    Whether to pick the next element of `a` (true) or `b` (false).
+ - `fa(ak)`, `fb(bk)`: Functions to apply to the picked elements
+
+```jldoctest
+julia> collect(imerge(1:2:5, 2:2:6, <=, identity, x->-x))
+6-element Vector{Int64}:
+  1
+ -2
+  3
+ -4
+  5
+ -6
+```
+"""
+imerge(a,b, p = <=, fa = identity, fb = identity) = IMerge(a,b,p,fa,fb)
```

Comment:
It seems like it might be good if this was:
```suggestion
imerge(a,b; fa = identity, fb = identity) = IMerge(a, b, <=, fa, fb)
imerge(p, a,b; fa = identity, fb = identity) = IMerge(a, b, p, fa, fb)
```
So we follow the julia convention of function first